examples/peer/migrations/2_deploy_contracts.js

```diff
@@ -1,12 +1,16 @@
 const Peer = artifacts.require('Peer')
+const PeerFactory = artifacts.require('PeerFactory')
 const Swap = artifacts.require('@airswap/swap/contracts/Swap.sol')
 const Types = artifacts.require('@airswap/types/contracts/Types.sol')
 
+const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000'
+
 module.exports = deployer => {
   deployer.deploy(Types)
   deployer.link(Types, Swap)
   deployer
     .deploy(Swap)
     .then(() => Swap.deployed())
-    .then(() => deployer.deploy(Peer, Swap.address))
+    .then(() => deployer.deploy(Peer, Swap.address, EMPTY_ADDRESS))
+  deployer.deploy(PeerFactory)
```

Comment:
can we not do a promise chain and instead deploy line by line like we had been doing in CREV2:https://github.com/airswap/factora-creV2/blob/master/migrations/2_base_migration.js

Reply:
Yeah sure, I prefer line by line too in migrations. However this is the case in all migrations for consumer, wrapper, peer, indexer and market, so it should be a separate PR/issue to tackle.